lib/mock/snapshot-agent.js

```
@@ -80,6 +82,20 @@ class SnapshotAgent extends MockAgent {
     handler = WrapHandler.wrap(handler)
     const mode = this[kSnapshotMode]
 
+    // Check if URL should be excluded (pass through without mocking/recording)
+    if (this[kSnapshotRecorder].isUrlExcluded(opts)) {
+      if (this[kRealAgent]) {
+        return this[kRealAgent].dispatch(opts, handler)
+      }
+      // If no real agent in playback mode, this is a configuration error
+      const error = new UndiciError('Cannot pass through excluded URL: no real agent available')
```

I'm not sure we should throw, maybe just go with no-op?

---

Sure, my thought was since this would likely be because of misconfiguration, we would want to fail loudly rather than have the request swallowed. Thoughts?

Yeah I noticed those as well. My thinking was these were unrelated but can we confirm?

---

Added some constructor logic in an earlier commit ensuring an agent is created if needed so we should be good. Removed the error block